src/parser/content-evaluator-module.ts

```diff
@@ -159,4 +204,15 @@ export class ContentEvaluatorModule implements Module {
       comments.length
     }.`;
   }
+
+  _generatePromptForReviewComments(issue: string, comments: ReviewCommentToEvaluate[]) {
+    if (!issue?.length) {
+      throw new Error("Issue specification comment is missing or empty");
+    }
+    return `I need to evaluate the value of a GitHub contributor's code review. Specifically, I'm interested in how much each code review comment helps to solve the GitHub issue and improve code quality. Please provide a float between 0 and 1 to represent the value of the code review comment. A score of 1 indicates that the comment is very valuable and significantly improves the submitted solution and code quality, whereas a score of 0 indicates a  negative or zero impact. A stringified JSON is given below that contains the specification of the GitHub issue, and code review comments by different contributors. The property "diff_hunk" presents the chunk of code being addressed for a possible change. \n\n\`\`\`\n${JSON.stringify(
+      { specification: issue, comments: comments }
```

Comment:
I am not sure if that makes sense. The idea was the pull-request comments are related to the code itself, and not to the specification which is why we put the relevance to `1` in the first place, because these comments are unrelated.

@0x4007 maybe the comments for the pull-request should be checked against the code block they are commenting, or something related to code anyway? I also usually like to specify in the pull-request what are the changes implied which also could be used instead of the specification.

Reply:
Specification is just for extra information. It will be helpful sometime and sometimes it will not be. We can remove it. But I feel replacing issue specification with PR specification will have same effect. When you specify a change in PR, it is broken down over several files, we are not sending all those files. We are just sending the few lines of code being addressed.  Plus, you will have to enforce similar practice to specify the changes in PR specification if you want to replace issue specification with PR specification.

Reply:
I tried without specifications. And there isn't a huge difference in evaluation by openai. It is reduced a little in one comment and increased a little in another. Here you can see code review comments evaluations [with specifications](https://github.com/EresDevOrg/ubiquibot-issues/issues/15#issuecomment-2294408525) and [without specifications](https://github.com/EresDevOrg/ubiquibot-issues/issues/15#issuecomment-2294428705). So, why waste tokens and I am keeping it without specification in this PR.

Reply:
I think include the specification because as a human reviewer sometimes (but not always) review the spec when looking at some of the changes. Maybe this is 1 out of 10 times but if I'm confused on why certain changes are made then I review the spec. I think it's a good idea to give the LLM the additional context if it's needed.

Reply:
I did a cost analysis on [this issue](https://github.com/ubiquibot/conversation-rewards/issues/5) including its PR. I will consider it a small to medium size issue. It costs 20 cents for both issue comment evaluation and PR comments evaluation. At the time of analysis, there was no specification in PR though. I will add it back now.